### ISSUE TITLE
free_bcc_memory() fails and returns 1 in a c++ app.

### ISSUE THREAD

**Issue by teroz** [iovisor/bcc#2172]
I just compiled the c++ examples in this repo having upgraded my libbcc to ver 0.8 on ubuntu 18.04.1 with the uname output as below.

_Linux wolverine 4.15.0-43-generic #46-Ubuntu SMP Thu Dec 6 14:45:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux_

the c++ application I compiled and run is ReadRandom.cc

**Comment by yonghong-song** [comment]
Could you check whether adding the following patch
```
commit dd132421c9edadc6cf8316e70454bedce6830155
Author: Sandipan Das <31861871+sandip4n@users.noreply.github.com>
Date:   Thu Jan 17 23:11:01 2019 +0530

    Fix process map parsing when freeing bcc memory (#2151)
    
    This fixes the format string used to parse the major and minor
    device fields in /proc/self/maps. These fields have hexadecimal
    values and hence cannot be parsed as unsigned integers.
    
    Fixes: 51480d05 ("implement free_bcc_memory() API (#2097)")
    Reported-by: Nageswara R Sastry rnsastry@linux.vnet.ibm.com
    Signed-off-by: Sandipan Das <sandipan@linux.ibm.com>
```
fixed the problem or not?

You can just try to build from the latest trunk as well.

**Comment by teroz** [comment]
works super with latest truck @yonghong-song  - thank you
